### PR TITLE
fix: cache promises instead of highlighters

### LINKED ISSUE #35: OOMs on the Jest docusaurus site
Maybe a leak in Shiki?

Screen Shot 2021-06-17 at 8 27 26 PM

### PR DESCRIPTION
**Pull request by frencojobs**
Closes #35

### CODE REVIEW

File: packages/remark-shiki-twoslash/src/index.ts [hunk 1]
```diff
@@ -95,7 +93,7 @@ const amendSettingsForDefaults = (settings: UserConfigSettings) => {
 
 const parsingNewFile = () => includes.clear()
 
-////////////////// The Remark API
```
Comment (frencojobs) [inline packages/remark-shiki-twoslash/src/index.ts:98]: The old comment was triggering some weird highlighting in my editor so I updated it even though it was unnecessary.

**Comment by frencojobs** [comment]
This fix makes it take only 30 seconds to build 123 files with a single simple twoslash code block which the current version fails to do at 80 files max.

**Review by orta — APPROVED** [review]
Clever

**Comment by orta** [comment]
Looks like a great solution - well done!

**Comment by orta** [comment]
Deployed as 1.5.0

**Pull request merged**